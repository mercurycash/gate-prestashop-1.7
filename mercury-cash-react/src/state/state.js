/* eslint-disable react-hooks/rules-of-hooks */
import { useState } from 'react';

import cryptocurrencies from "../configs/cryptocurrencies";


export const initialState = {
	step: 1,
	createTransactionStatus: false,
	cryptoButtons: cryptocurrencies
}

export const store = () => {
	const [ state, setState ] = useState(initialState)

	// Write initialization to state
	const initData = (data) => setState(prevState => ({ ...prevState, initializedData: data }))

	// Change modal steps
	const changeStep = (step) => setState(prevState => ({ ...prevState, step: step }))

	// Transaction status
	const createTransactionStatus = (bool) => setState(prevState => ({
		...prevState,
		createTransactionStatus: bool
	}))

	// Write checkout data in state from checkoutQuery
	const writeTransactionData = (obj) => setState(prevState => ({ ...prevState, transactionData: obj }))

	// write transaction status in transactionData
	const writeTransactionStatus = (status) => setState(prevState => ({
		...prevState,
		transactionData: { ...prevState?.transactionData, status: status }
	}))
	
	// Set confirmation
	const setConfirmation = (confirmations) => setState(prevState => ({
		...prevState,
		transactionData: { ...prevState?.transactionData, confirmations: confirmations }
	}))

	// Update confirmations status
	const updateStatusConfirmation = (data) => setState(prevState => ({
		...prevState, transactionData: {
			...prevState.transactionData,
			confirmations: data
		}
	}))
  
  // error
  const errorMessage = (data) => setState(prevState => ({
    ...prevState,
    error: data
  }))

	return {
		state,
		changeStep,
		createTransactionStatus,
		initData,
		writeTransactionData,
		updateStatusConfirmation,
		writeTransactionStatus,
		setConfirmation,
    errorMessage
	}
}