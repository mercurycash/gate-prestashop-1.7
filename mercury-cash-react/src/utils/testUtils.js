import { render } from "@testing-library/react";
import { Provider } from "outstated";
import { store } from "../state/state";

export const customRender = (ui) => {
  return render(
    <Provider stores={ [ store ] }>{ ui }</Provider>
  )
}
