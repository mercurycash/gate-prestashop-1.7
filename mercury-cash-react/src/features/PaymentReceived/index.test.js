import { PaymentReceived } from "./index";
import { customRender } from "../../utils/testUtils";
import { initialState, store } from "../../state/state";
import { renderHook, act } from '@testing-library/react-hooks'

describe('<PaymentReceived/>', () => {
  const spy = jest.spyOn(global.console, "error")

  it('Payment component render and set checkout data in state', () => {
    let writeTransactionData, state;
    renderHook(() => ({
      writeTransactionData, state
    } = store()))
    expect(state).toBe(initialState)

    act(() => {
      writeTransactionData({
        cryptoAmount: 0.04230942,
        confirmations: 3,
        address: '0x521273d0a5b93fbb4001c40b12d88e632ebeee8b',
        qrCodeText: 'SomeQRCode',
        exchangeRate: '1,182.77',
        networkFee: '0.000102891241761',
        uuid: 'someUUID',
        cryptoCurrency: 'ETH'
      })
    })

    const { container } = customRender(<PaymentReceived/>)
    expect(container).toMatchSnapshot()
  })
})