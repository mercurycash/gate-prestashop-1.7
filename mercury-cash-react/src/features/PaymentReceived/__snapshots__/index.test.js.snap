// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PaymentReceived/> Payment component render and set checkout data in state 1`] = `
<div>
  <h3
    class="mercury-cash__title mercury-cash__title--big"
  >
    received
  </h3>
  <h4
    class="mercury-cash__text mercury-cash__text--purple"
  >
    dontCloseWindow
  </h4>
  <img
    alt=""
    class="mercury-cash__gif"
    src="preloader.gif"
  />
  <div
    class="mercury-cash__timer mercury-cash__timer--uppercase"
  >
    confirmations
    : 
  </div>
  <h4
    class="mercury-cash__text "
  >
    pleaseWaitConfirmation
  </h4>
</div>
`;
